src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -313,39 +315,23 @@ doGetBlockParent v cid dbenv (bh, hash)
     qtext = "SELECT hash FROM BlockHistory WHERE blockheight = ?"
 
 
-doRegisterSuccessful :: Db logger -> PactHash -> IO ()
-doRegisterSuccessful dbenv (TypedHash hash) =
-    runBlockEnv dbenv (indexPactTransaction $ BS.fromShort hash)
-
-doLookupSuccessful :: Db logger -> Maybe ConfirmationDepth -> V.Vector PactHash -> IO (HashMap.HashMap PactHash (T2 BlockHeight BlockHash))
-doLookupSuccessful dbenv confDepth hashes = runBlockEnv dbenv $ do
+doLookupSuccessful :: BlockHeight -> V.Vector PactHash -> BlockHandler logger SQLiteEnv (HashMap.HashMap PactHash (T2 BlockHeight BlockHash))
+doLookupSuccessful curHeight hashes = do
     withSavepoint DbTransaction $
       fmap buildResultMap $ -- swizzle results of query into a HashMap
       callDb "doLookupSuccessful" $ \db -> do
-        -- if there is a confirmation depth, we get the current height and calculate
-        -- the block height, to look for the transactions in range [0, current block height - confirmation depth]
-        blockheight <- case confDepth of
-          Nothing -> pure Nothing
-          Just (ConfirmationDepth cd) -> do
-            let currentHeightQ = "SELECT blockheight FROM BlockHistory ORDER BY blockheight DESC LIMIT 1"
-            currentHeight <- qry_ db currentHeightQ [RInt]
-            case currentHeight of
-              [[SInt bh]] -> pure $ Just (bh - fromIntegral cd)
-              _ -> fail "impossible"
-
         let
           hss = V.toList hashes
           params = Utf8 $ intercalate "," (map (const "?") hss)
           qtext = "SELECT blockheight, hash, txhash FROM \
                   \TransactionIndex INNER JOIN BlockHistory \
                   \USING (blockheight) WHERE txhash IN (" <> params <> ")"
-                  <> maybe "" (const " AND blockheight <= ?") confDepth
-                  <> ";"
-          qvals 
+                  <> " AND blockheight <= ?;"
+          qvals
             -- match query params above. first, hashes
             = map (\(TypedHash h) -> SBlob $ BS.fromShort h) hss
-            -- then, the block depth, if any
-            ++ maybe [] (\bh -> [SInt bh]) blockheight
+            -- then, the block height
```

Comment:
Maybe add a comment the `-1`. Or not. I think it's fine and this makes sense after thinking about it